docs/conceptual/cryptography/session-keys.md

```diff
@@ -4,25 +4,34 @@ title: Session Keys
 
-You can declare any number Session keys. For example, the default Substrate node uses three. Other chains could have more or fewer depending on what operations the chain expects its validators to perform.
+In practice, validators amalgamate all of the session public keys into a certificate by signing the set of public keys with a "Controller" account and submitting a transaction. This on-chain certificate links a validator node with an account that holds funds. As such, that account can be credited with rewards or slashed based on the node's behavior.
+
+The runtime declares what session keys will be included in a certificate (`runtime/src/lib.rs`):
 
 ```rust
 pub struct SessionKeys {
-  #[id(key_types::GRANDPA)]
-  pub grandpa: GrandpaId,
-  #[id(key_types::BABE)]
-  pub babe: BabeId,
-  #[id(key_types::IM_ONLINE)]
-  pub im_online: ImOnlineId,
+    pub grandpa: Grandpa,
+    pub babe: Babe,
+    pub im_online: ImOnline,
+    pub authority_discovery: AuthorityDiscovery,
+}
+```
+
+The actual cryptographic curve that each key uses gets defined in `primitives`. For example, BABE's key uses sr25519:
+
+```rust
+mod app {
+	  use sp_application_crypto::{app_crypto, key_types::BABE, sr25519};
+	  app_crypto!(sr25519, BABE);
 }
 ```
 
-> **Note:** This code is just an example of the Substrate node at the time of writing. Refer to the runtime for the most up-to-date implementation.
+> **Note:** This code is just an example of the Substrate node at the [time of writing](https://github.com/paritytech/substrate/tree/9fa8589d9b8cfe8716e9e4c48f9e3f238c1e502f). Refer to the runtime for the most up-to-date implementation.
 
-The default Substrate node implements Session keys in the [Session module](https://substrate.dev/rustdocs/master/pallet_session/).
+The default Substrate node implements Session keys in the [Session pallet](https://substrate.dev/rustdocs/master/pallet_session/).
 
 ## Generation and Use
 
-Session keys are hot keys that must be kept online. They are not meant to be used as account keys. If one of the Session keys is compromised, the attacker could commit slashable behavior. Session keys may be changed regularly (e.g. every session) via [RPC](https://substrate.dev/rustdocs/master/sc_rpc/author/trait.AuthorApi.html#tymethod.rotate_keys) for increased security.
+Session keys are hot keys that must be kept online. The individual keys should **not** be used to control funds, and the keys should be treated as a single certificate rather than as individual keys. All the logic for handling them is in the Substrate client, primitives, and Session pallet. If one of the Session keys is compromised, the attacker could commit slashable behavior. Session keys may be changed regularly (e.g. every session) via [RPC](https://substrate.dev/rustdocs/master/sc_rpc/author/trait.AuthorApi.html#tymethod.rotate_keys) for increased security.
```

Comment:
"as a single certificate"

I always find this confusing to call this stuff certificates. I'm also not sure what exactly you mean by this.